internal/olm/operator/registry/operator_installer.go

```diff
@@ -122,48 +124,66 @@ func (o OperatorInstaller) waitForCatalogSource(ctx context.Context, cs *v1alpha
 	return nil
 }
 
-// createOperatorGroup creates an OperatorGroup using package name if an OperatorGroup does not exist.
-// If one exists in the desired namespace and it's target namespaces do not match the desired set,
-// createOperatorGroup will return an error.
-func (o OperatorInstaller) createOperatorGroup(ctx context.Context) error {
-	targetNamespaces := make([]string, len(o.InstallMode.TargetNamespaces), cap(o.InstallMode.TargetNamespaces))
-	copy(targetNamespaces, o.InstallMode.TargetNamespaces)
+func (o OperatorInstaller) ensureOperatorGroup(ctx context.Context) error {
 	// Check OperatorGroup existence, since we cannot create a second OperatorGroup in namespace.
 	og, ogFound, err := o.getOperatorGroup(ctx)
 	if err != nil {
 		return err
 	}
-	// TODO: we may need to poll for status updates, since status.namespaces may not be updated immediately.
-	if ogFound {
-		// targetNamespaces will always be initialized, but the operator group's namespaces may not be
-		// (required for comparison).
-		if og.Status.Namespaces == nil {
-			og.Status.Namespaces = []string{}
-		}
-		// Simple check for OperatorGroup compatibility: if namespaces are not an exact match,
-		// the user must manage the resource themselves.
-		sort.Strings(og.Status.Namespaces)
-		sort.Strings(targetNamespaces)
-		if !reflect.DeepEqual(og.Status.Namespaces, targetNamespaces) {
-			msg := fmt.Sprintf("namespaces %+q do not match desired namespaces %+q", og.Status.Namespaces, targetNamespaces)
-			if og.GetName() == operator.SDKOperatorGroupName {
-				return fmt.Errorf("existing SDK-managed operator group's %s, "+
-					"please clean up existing operators `operator-sdk cleanup` before running package %q", msg, o.PackageName)
+
+	supported := o.SupportedInstallModes
+
+	// --install-mode was given
+	if !o.InstallMode.IsEmpty() {
+		if o.InstallMode.InstallModeType == v1alpha1.InstallModeTypeSingleNamespace {
+			targetNsSet := sets.NewString(o.InstallMode.TargetNamespaces...)
+			if !supported.Has(string(v1alpha1.InstallModeTypeOwnNamespace)) && targetNsSet.Has(o.cfg.Namespace) {
+				return fmt.Errorf("cannot watch namespace %q: operator %q does not support install mode %q", o.cfg.Namespace, o.StartingCSV, v1alpha1.InstallModeTypeOwnNamespace)
```

Comment:
Is this check necessary?

I feel like the error message could be confusing. IIUC, a user will see this if:
1. the operator CSV supports SingleNamespace, but not OwnNamespace.
2. the command line options include `--install-mode=SingleNamespace=my-namespace` and  `--namespace=my-namespace` (or if `my-namespace` is setup with the current context)

If I specified `SingleNamespace` and then see an error message telling my that `OwnNamespace` is not supported, I would probably file a bug because I asked for SingleNamespace, not OwnNamespace.

I think it might be more intuitive for users if we check if `--install-mode=SingleNamespace=my-namespace` would be equivalent to `OwnNamespace` and return an error saying they should use `OwnNamespace` instead. Looks like that's what you have below in lines 144-147

Reply:
I think I need to actually move this to CheckCompatibility.